assets/js/blocks/product-collection/inspector-controls/index.tsx

```diff
@@ -102,3 +106,23 @@ const ProductCollectionInspectorControls = (
 };
 
 export default ProductCollectionInspectorControls;
+
+export const withUpgradeNoticeControls =
+	< T extends EditorBlock< T > >( BlockEdit: ElementType ) =>
```

Comment:
The generic type parameter declaration syntax still looks a bit exotic to me 😅

Kudos @kmanijak for your efforts ensuring our code is always TS valid.